src/main/java/org/jmxtrans/agent/StatsDOutputWriter.java

```diff
@@ -131,6 +131,14 @@ public void writeInvocationResult(String invocationName, Object value) throws IO
     @Override
     public synchronized void writeQueryResult(String metricName, String metricType, Object value) throws IOException
     {
+        // statsd expects a number value for the metric.
+        //
+        // skip if value's string representation equals to "NaN" or "INF", which are meaningless values to statsd.
+        // passing the invalid values down will trigger error in downstream parsing applications.
+        if (String.valueOf(value).equals("NaN") || String.valueOf(value).equals("INF")) {
```

Comment:
It's better to do a value != value for the NaN check

Reply:
the thing is we also need to check if the value is "INF".